examples/CMakeLists.txt

```diff
@@ -2,23 +2,23 @@ project("kafka-examples")
 
 # Target: kafka_sync_producer
 add_executable("kafka_sync_producer" "kafka_sync_producer.cc")
-target_link_libraries("kafka_sync_producer" modern-cpp-kafka-api)
+target_link_libraries("kafka_sync_producer" modern-cpp-kafka-api ${SASL_LIBRARY})
```

Comment:
If `${SASL_LIBRARY}` is not specified, it will be empty and it won't affect existing usage.